lib/finder.js

```diff
@@ -0,0 +1,208 @@
+
+    // when we have a single line comment
+    if (singleLineComment > -1 && !isInComment){
+      line = line.substr(0, singleLineComment);
+      continue;
+    }
+
+    // when there's no comment chars
+    if (startIndex === -1 && endIndex === -1) {
+      return { 
+        line: isInComment ? "" : line, 
+        isInComment: isInComment
+      };
+    }
+
+    // when there's a start and end
+    if (startIndex > -1 && endIndex > -1) { 
+      line = line.substr(0, startIndex) + line.substr(endIndex + 2);
+      continue;
+    }
+
+    // when there's a start, but no end
+    if (startIndex > -1 ) return { line : line.substr(0, startIndex), isInComment: true };
+    
+    // when there's an end, but no start
+    if (endIndex > -1 && isInComment) return { line: line.substr(endIndex + 2), isInComment: false };
+  } 
+
+  return { line: "", isInComment: isInComment };
+}
+
+var splitExposedFunctions = function(exposingLine) {
+  return exposingLine
+    .substr(0, exposingLine.lastIndexOf(')'))
+    .split(',')
+    .map((str) => str.trim())
+    .filter((str) => str[0].toLowerCase() === str[0]);
+};
+
+var isAModuleLine = function(line){
+  return line.startsWith('module ')
+    || line.startsWith('port module')
+    || line.startsWith('effect module');
+};
+
+
+function Parser(){
+  // if we're currently in a comment
+  var isInComment = false;
+
+  // if the file does not have a module line
+  var isMissingModuleName = false;
+
+  // if we're done parsing
+  var parsingDone = false;
+  
+  // if the module line has been read
+  var hasModuleLineBeenRead = false;
+
+  var isReadingModuleName = false;
+  var isReadingExports = false;
+  var isBetweenBrackets = false;
+
+  // functions that have been exposed
+  var exposedFunctions = [];
+  // number of open/closed brackets seen so far
+  var openBracketsSeen = 0;
+  var closedBracketsSeen = 0;
+  // data between exposing brackets
+  var data = "";
+
+
+  this.parseLine = function(line){
+    if (parsingDone) return;
+    
+    var whereWeUpTo = stripComments(line, isInComment);
+    isInComment = whereWeUpTo.isInComment;
+    line = whereWeUpTo.line.trim();
+
+    if (line.length == 0) return;
+
+    // if we haven't started reading the first line
+    if (!hasModuleLineBeenRead && isAModuleLine(line)) {
+      hasModuleLineBeenRead = true;
+      // drop module from the line
+      line = line.substr(line.indexOf('module') + 7);
+      isReadingModuleName = true;
+
+      if (line.length === 0) return;
+    } 
+
+    // if we manage to find content before the module line,
+    // something is wrong - so exit
+    if (!hasModuleLineBeenRead) {
+      isMissingModuleName = true;
+      parsingDone = true;
+      return;
+    }
+
+    // if we are still reading the module line
+    if (isReadingModuleName) {
+      var exposingIndex = line.indexOf('exposing');
+
+      // if we haven't found exposing yet
+      if (exposingIndex === -1) {
+        if (line.length > 0) parsingDone = true;
```

Comment:
This seems a little weird, and I think it's why #138 is occurring. @eeue56 could you clarify in which case this is required?